ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -164,7 +166,7 @@ def _not_done_training(self) -> bool:
         return (
             any([t.get_step <= t.get_max_steps for k, t in self.trainers.items()])
             or not self.train_model
-        )
+        ) or len(self.trainers) == 0
```

Comment:
Could we end up in a situation where the environment connects but never registers a brain?

Reply:
Yes, but nothing would happen. You would call reset on it and it would just hang forever. Just like when there are no learning brains in the scene currently.